src/modules/accounting/accounting.py

```diff
@@ -0,0 +1,210 @@
+                block_identifier=blockstamp.block_hash
+            ),
+            block_identifier=blockstamp.block_hash,
+        ))
+
+    def _get_last_withdrawal_request_to_finalize(self, blockstamp: BlockStamp) -> int:
+        chain_config = self._get_chain_config(blockstamp)
+        frame_config = self._get_frame_config(blockstamp)
+
+        is_bunker = self._is_bunker(blockstamp)
+        withdrawal_vault_balance = self._get_withdrawal_balance(blockstamp)
+        el_rewards_vault_balance = self._get_el_vault_balance(blockstamp)
+        finalization_share_rate = self._get_finalization_shares_rate(blockstamp)
+
+        withdrawal_service = Withdrawal(self.w3, blockstamp, chain_config, frame_config)
+
+        return withdrawal_service.get_next_last_finalizable_id(
+            is_bunker,
+            finalization_share_rate,
+            withdrawal_vault_balance,
+            el_rewards_vault_balance,
+        )
+
+    @lru_cache(maxsize=1)
+    def _get_finalization_shares_rate(self, blockstamp: BlockStamp) -> int:
+        simulation = self.get_rebase_after_report(blockstamp)
+        return simulation.post_total_pooled_ether * SHARE_RATE_PRECISION_E27 // simulation.post_total_shares
+
+    @lru_cache(maxsize=1)
+    def get_rebase_after_report(self, blockstamp: BlockStamp) -> LidoReportRebase:
+        chain_conf = self._get_chain_config(blockstamp)
+        frame_config = self._get_frame_config(blockstamp)
+
+        last_ref_slot = self.report_contract.functions.getLastProcessingRefSlot().call(
+            block_identifier=blockstamp.block_hash,
+        )
+
+        if not last_ref_slot:
+            slots_elapsed = blockstamp.ref_slot - frame_config.initial_epoch * chain_conf.slots_per_epoch
+        else:
+            slots_elapsed = blockstamp.ref_slot - last_ref_slot
+
+        validators_count, cl_balance = self._get_consensus_lido_state(blockstamp)
+
+        timestamp = chain_conf.genesis_time + blockstamp.ref_slot * chain_conf.seconds_per_slot
+
+        result = self.w3.lido_contracts.lido.functions.handleOracleReport(
+            timestamp,  # _reportTimestamp
+            slots_elapsed * chain_conf.seconds_per_slot,  # _timeElapsed
+            validators_count,  # _clValidators
+            Web3.to_wei(cl_balance, 'gwei'),  # _clBalance
+            self._get_withdrawal_balance(blockstamp),  # _withdrawalVaultBalance
+            self._get_el_vault_balance(blockstamp),  # _elRewardsVaultBalance
+            0,  # _lastFinalizableRequestId
+            0,  # _simulatedShareRate
+        ).call({'from': self.w3.lido_contracts.accounting_oracle.address})
+
+        return LidoReportRebase(*result)
+
+    def _is_bunker(self, blockstamp: BlockStamp) -> bool:
+        frame_config = self._get_frame_config(blockstamp)
+        chain_config = self._get_chain_config(blockstamp)
+        rebase_report = self.get_rebase_after_report(blockstamp)
```

Comment:
we can't reuse it in bunker since it contains execution layer rewards. need another static call without el rewards